File: 2. Loop Problems/Loop Problems in c/p10.c
// Write a C program to find frequency of each digit in a given integer

#include <stdio.h>

int main()
{
    int n, temp = 0, count;
    printf("Enter number: ");
    scanf("%d", &n);

    

    for (int digit = 0; digit <= 9; digit++)
    {

        temp = n;
        count = 0;

        while (temp != 0)
        {
            if (temp % 10 == digit)
            {
                count++;
            }
            temp = temp / 10;
        }

        if (count > 0)
        {
            printf("%d is here %d times.\n", digit, count);
        }
    }
}

File: 2. Loop Problems/Loop problems in python/p10.py
# Python program to find the frequency of each digit in a given integer

n = int(input("Enter a number: "))
n = abs(n)  # Ensure n is positive in case of negative input

# Loop through digits from 0 to 9
for digit in range(10):
    count = 0
    temp = n  # Copy of n to work with

    # Count the occurrences of the current digit
    while temp != 0:
        if temp % 10 == digit:
            count += 1
        temp = temp // 10  # Remove the last digit

    # Print the frequency if the digit occurs
    if count > 0:
        print(f"Digit {digit} appears {count} times.")
